### ISSUE TITLE
Add new model code

### ISSUE THREAD

**Issue by zparis99** [hassonlab/podcast-benchmark#14]
We need to add a model type which contains both a trainable brain adapter (like in the paper) which maps from brain data to input embeddings and also contains a frozen LLM. We can take some inspiration from the original paper code: https://github.com/YeZiyi1998/Brain-language-generation/blob/v1.0.0/language_generation/src/top_model_utils.py although we'll likely need to modify it to make this fit well.

It would need to take as input our windowed neural data and the token context. It also needs two modes:

1. We return only the brain adapter output so we can do the warm up training from the paper that maps the brain adapter to the mean of the context tokens.
2. Another which returns logits over all of the possible output tokens which we use for the final steps of training.

For now let's freeze the LLM weights because that's what they do in the paper and trying to train the LLM would open a whole other can of worms that will make life harder (we'd likely want to use some other training library than vanilla pytorch).
